### ISSUE TITLE
Custom module with Many-to_One relationship to Users does not show Users in listview

### ISSUE THREAD

**Issue by torinfo** [salesagility/SuiteCRM-Core#499]
### Issue

I have a custom module (created in 7.11 before migration to 8.6) that has a Many-to-One relationship to Users

It doesn't show the Users name in the listview, even though that field is included.
It works fine in the Detail/Edit views of the custom module.

Also (but I don't know whether that is really related) the Edit/Detail view of a user does not work anymore!!

As a test I created a basic test module with just a relationship to Users in 8.6. It shows the same issue. I've included the published custum test module.

[test2024_06_07_105829.zip](https://github.com/user-attachments/files/15728443/test2024_06_07_105829.zip)

### Steps to Reproduce the Issue

```bash
1. Create custom module, choose basic
2. Add a Many to One relationship to Users module
3. Add the User to the listview
4. Create a record and navigate to the list.
Theuser column is empty.
...
```

### Version

8.6.0

### What browser are you currently using?

Chrome

### Browser Version

Version 125.0.6422.77 (Official Build) (64-bit)

### Environment Information

MariaDB 10.6.16, PHP 8.1.2

### Operating System and Version

Ubuntu 22.04.4 LTS

**Comment by scicali** [comment]
duplicate #477

**Comment by scicali** [comment]
#489 may fix it, in next relase

**Comment by torinfo** [comment]
I implemented the change, but it doesn't seem to work. As an extra check I added some logging, but I don't see anything being logged. I tried Quick Repair and also Repair Relationships, but that doesn't show anything either.

**Comment by scicali** [comment]
Have you implemented backend part also?
`
if ($isNonDb === true && empty($fieldDefinition['id_name'])) {
                return false;
            }
`

this is my temporary workaround (implemented before the fix)

`
            if ($isNonDb === true) {
                //return false;
            }
`

**Comment by torinfo** [comment]
Yes I did and as far as I can tell (because of the added logging) that function is never called in my situation.

**Comment by scicali** [comment]
Maybe "relationship" is handle in different way than "relate field"

In "Steps to Reproduce the Issue", point 2, I add a relate field to users module, not a relationship. So the issue is slightly different.

**Comment by clemente-raposo** [comment]
Hi @torinfo @scicali 

As @scicali mentioned it seems to us that this is related with: https://github.com/salesagility/SuiteCRM-Core/issues/477.

The PR https://github.com/salesagility/SuiteCRM-Core/pull/489, has been merged and released on 8.6.1.

We've re-tested by creating a new custom module using studio and adding both the releationship described above as well as the relate field scenario. It seems to work as expected ( see the following gif). 

Please consider upgrading. If the issue still occurs after the upgrade, please re-open this issue.

Closing the issue.

![custom_module_user_relationship_demo](https://github.com/salesagility/SuiteCRM-Core/assets/1140849/261b2249-fb15-4d7d-b91f-e97cbb2942a6)

**Comment by torinfo** [comment]
@clemente-raposo I can confirm that after upgrade it works as expected. Thank you!
